### ISSUE TITLE
Rename server

### ISSUE THREAD

**Issue by lliming** [XSEDE/globus-ssh-server#60]
Rename the server to something more generically OAuth2.

- Rename PAM module from `pam_globus.so` to something more generically OAuth2, such as `pam_oauth2.so`.
- Rename this GitHub repo.

**Comment by JasonAlt** [comment]
It would be misleading to relabel the client or server components as 'oauth' because these are not generically oauth compliant. These were designed to work with the Globus Auth security model, however it evolves.

**Comment by lliming** [comment]
I don't believe the last comment is correct. The tokens that are passed from client to server, the service interface the client uses to obtain tokens, and the introspection interface the server uses are 100% compliant with the OAuth2 spec (plus OIDC 1.0), and those are the central features of this code.

It's true that the code goes farther and uses extra data that Globus adds to the introspection response to do more than the minimum that OAuth2 requires. This doesn't violate or invalidate OAuth2 compliance, though, because OAuth2 doesn't say there can't be extra data in the introspection response. (RFC 7662, section 2.2. "Specific implementations MAY extend this structure with their own service-specific response names as top-level members of this JSON object.") 

Further, we need to distinguish between what the code does now vs. what it can/will do in the future. Although the initial implementation is based on Globus's Auth service, Globus won't be the only developers for this code moving ahead. There's nothing preventing the developer community from extending the code (perhaps making the Globus parts conditional) so it can be used with other OAuth2 services.  What makes this code distinct and valuable, and is unlikely to change, is that it provides an OAuth2-based authorization method for SSH.
